src/Bicep.RegistryModuleTool/Extensions/TempFile.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO.Abstractions;
+
+namespace Bicep.RegistryModuleTool.Extensions
+{
+    internal sealed class TempFile : IDisposable
+    {
+        private readonly IFileSystem fileSystem;
+
+        private string? path;
+
+        public TempFile(IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+            this.path = fileSystem.Path.GetTempFileName();
+        }
+
+        public string Path => this.path ?? throw new ObjectDisposedException(this.GetType().Name);
+
+        ~TempFile() => this.DisposeInternal();
+
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+
+            this.DisposeInternal();
+        }
+
+        private void DisposeInternal()
+        {
+            if (path is null)
+            {
+                // Already disposed.
+                return;
+            }
+
+            try
+            {
+                this.fileSystem.File.Delete(path);
```

Comment:
>Delete [](http://example.com/codeflow?start=37&length=6)

Btw, DeleteOnClose is supported on [https://docs.microsoft.com/en-us/dotnet/api/system.io.fileoptions?view=net-6.0](https://docs.microsoft.com/en-us/dotnet/api/system.io.fileoptions?view=net-6.0). Not sure if it'll fit, but one of the FileStream overloads allows you pass that in. On Windows it's implemented by the OS. On Linux and Mac, FileStream does a delete during Dispose().

Reply:
Let the OS handle the deletion is definitely better (even it is Windows only). Removed the custom temp file and changed the implementation to use the `DeleteOnClose`.